File: internal/models/node.go
package models

// NodeType represents the type of a node in the explanation graph.
type NodeType string

const (
	NodeTypeInput    NodeType = "input"
	NodeTypeComputed NodeType = "computed"
	NodeTypeOutput   NodeType = "output"
	NodeTypeMissing  NodeType = "missing"
)

// Node represents a single node in the explanation graph.
type Node struct {
	ID         string            `json:"id"`
	Label      string            `json:"label"`
	Value      float64           `json:"value"`
	Confidence float64           `json:"confidence"`
	NodeType   NodeType          `json:"node_type"`
	Metadata   map[string]string `json:"metadata,omitempty"`
}

File: src/explainable_engine/models/node.py
from enum import Enum

from pydantic import BaseModel, Field


class NodeType(str, Enum):
    INPUT = "input"
    COMPUTED = "computed"
    OUTPUT = "output"
    MISSING = "missing"


class Node(BaseModel):
    id: str
    label: str
    value: float
    confidence: float = Field(ge=0.0, le=1.0, default=1.0)
    node_type: NodeType = NodeType.INPUT
    metadata: dict | None = None

    model_config = {
        "json_schema_extra": {
            "examples": [
                {
                    "id": "trend_strength",
                    "label": "Trend Strength",
                    "value": 0.8,
                    "confidence": 0.9,
                    "node_type": "input",
                }
            ]
        }
    }
